tower/src/lib.rs

```diff
@@ -1,14 +1,38 @@
 //! Various utility types and functions that are generally with Tower.
 
-pub mod builder;
 #[macro_use]
 extern crate futures;
-#[cfg(test)]
-extern crate tokio_mock_task;
+
 extern crate tower_layer;
 extern crate tower_service;
 extern crate tower_service_util;
 
+#[cfg(feature = "layers")]
+extern crate tower_buffer;
+#[cfg(feature = "layers")]
+extern crate tower_filter;
+#[cfg(feature = "layers")]
+extern crate tower_in_flight_limit;
+#[cfg(feature = "layers")]
+extern crate tower_load_shed;
+#[cfg(feature = "layers")]
+extern crate tower_rate_limit;
+#[cfg(feature = "layers")]
+extern crate tower_retry;
+
+#[cfg(feature = "make_service")]
+extern crate tower_balance;
+#[cfg(feature = "make_service")]
+extern crate tower_discover;
+#[cfg(feature = "make_service")]
+extern crate tower_reconnect;
+
+pub mod builder;
+#[cfg(feature = "layers")]
+pub mod layer;
+#[cfg(feature = "make_service")]
+pub mod make_service;
+pub mod service_util;
```

Comment:
I don't think we need this named `service_util`. The types that matter should be exported in `tower::util`. service-util will be renamed tower-util.